container.sh

```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env bash
+
+cd "$(dirname "${BASH_SOURCE[0]}")"
+
+ENGINE=docker
+
+if ! which "$ENGINE" &>/dev/null ; then
+    ENGINE=podman
+fi
+
+if ! which "$ENGINE" &>/dev/null ; then
+    echo "No docker or podman in PATH" >&2
+    exit 1
+fi
+
+read -d '' script <<EOF
+cd /app
+mkdir .nix
+cargo build
+
+run() {
+    cargo run .nix "\$@"
+}
+
+export -f run
+exec bash
+EOF
+
+"$ENGINE" run -v $PWD:/app -it rust bash -c "$script"
```

Comment:
I had to add this:

```suggestion
"$ENGINE" run -v $PWD:/app --privileged -it rust bash -c "$script"
```

Reply:
Ah. I'm using podman, which must have done stuff differently because for me it worked without --privileged.